### ISSUE TITLE
StackNavigator inside a TabNavigator causes a back button to be in header

### ISSUE THREAD

**Issue by benschel** [react-community/react-navigation#1498]
Here is my navigator structure:
- RootNav (StackNavigator)
  - TabsNav (TabNavigator)
    - MealsNav (StackNavigator)
      - MyMeals (Component)
    - FindRecipes (Component)

So there are two tabs, one of them has a screen that's a StackNavigator which has a screen that's a component. The other tab has a screen that is just a component.

```javascript
const mealsNavConfig = {
    routeConfig: {
        MyMeals: {
            screen: MyMeals,
            navigationOptions: {
                title: 'My Meals'
            }
        }
    }
};

export const MealsNav = StackNavigator(mealsNavConfig.routeConfig);


const tabsNavConfig = {
    routeConfig: {
        MealsNav: {
            screen: MealsNav,
            navigationOptions: {
                tabBarLabel: 'My Meals',
                tabBarIcon: ({ tintColor }) => (<Icon name="cutlery" size={25} color={tintColor} />)
            }
        },
        FindRecipes: {
            screen: FindRecipes,
            navigationOptions: {
                tabBarLabel: 'Find Recipes',
                tabBarIcon: ({ tintColor }) => (<Icon name="search" size={25} color={tintColor} />)
            }
        }
    }
}

export const TabsNav = TabNavigator(tabsNavConfig.routeConfig);


const rootNavConfig = {
    routeConfig: {
        Home: {
            screen: TabsNav
        }
    },
    navigatorConfig: {
        mode: 'modal',
        headerMode: 'none'
    }
};

export const RootNav = StackNavigator(rootNavConfig.routeConfig, rootNavConfig.navigatorConfig);
```

I'm using redux so my navReducer sets the initial state to be the MyMeals route.

```javascript
const initialRoute = 'MyMeals';
const initialState = RootNav.router.getStateForAction(NavigationActions.navigate({
    routeName: initialRoute
}));

export default (state = initialState, action) => {
    nextState = RootNav.router.getStateForAction(action, state);

    return nextState || state;
}
```

When the app loads the "My Meals" screen has a back button and when you tap it takes you back to another "My Meals" screen. It's like there is an extra stack. 

<img width="382" alt="screen shot 2017-05-13 at 9 26 28 pm" src="https://cloud.githubusercontent.com/assets/3179718/26030488/10d53050-3823-11e7-93ec-679d748477c0.png">

Also when you tap the "Find Recipes" tab it takes you to the "Find Recipes" screen but on the left edge there is a shadow. It looks like the shadow that you see when you are transitioning to a different screen. 

<img width="385" alt="screen shot 2017-05-13 at 9 26 51 pm" src="https://cloud.githubusercontent.com/assets/3179718/26030489/15d0b264-3823-11e7-8dd2-842c7efa3ad4.png">

The expected behavior is for there to not be a back button on "MyMeals" when the app first loads. And there not to be a shadow on the "Find Recipes" tab. 

Is this a bug, or am I doing something wrong.

`react-native: 0.44.0`
`react-navigation: 1.0.0-beta.9`
`redux: 3.6.0`
`react-redux: 5.0.4`
`yarn: 0.23.4`
`node: 7.10.0`

**Comment by AlanFoster** [comment]
@benschel Have you tried setting your initial route to `'MealsNav'` ?

**Comment by aqnaruto** [comment]
and in you screenshot the text is blur 
![image](https://cloud.githubusercontent.com/assets/9067159/26033369/553cfb00-38dd-11e7-865e-cff8a3445d67.png)
so,how to fix the text is blue duaring transaction-animation ? #1465

**Comment by benschel** [comment]
@AlanFoster Thanks! That does get rid of the back button. It seems to be working as it should. 

The shadow is still there on the other tab. That might be a bug. I imagine it will go away if I add a StackNavigator to the second tab. 

@aqnaruto For me the text in the screenshot and the simulator isn't blurry. The whole image you posted is blurry so I think it has to do with how you are viewing the screenshot, and not the actual application.

**Comment by zhulinpinyu** [comment]
same problem for me, did you got solution? @benschel

**Comment by zhulinpinyu** [comment]
Same issue for me, Did you get solution? @benschel

**Comment by eflath** [comment]
I'm seeing the shadow as well, with a similar setup (StackNavigators inside TabNavigator)

**Comment by spencercarli** [comment]
Is this still an issue in the latest release?

**Comment by Mephalrith** [comment]
yeah, I'm getting the same kind of issue with the most recent version

**Comment by NunoCardoso** [comment]
I am bootstrapping an app now, and I have the same issue. Seems like it's still there.

**Comment by NunoCardoso** [comment]
This is my TabNav class. It boots by showing a Search page with a back Search button, that goes to another Search page. 

const SearchTab = StackNavigator({
    Search: {
        screen: Search,
        path: '/',
        navigationOptions: ({navigation}) => ({
            title: I18n.t('search'),
        })
    }
},{
    cardStyle: {
        backgroundColor: 'white'

    }
});

const FavoritesTab = StackNavigator({
    Favorites: {
        screen: Favorites,
        path: '/',
        navigationOptions: ({navigation}) => ({
            title: I18n.t('favorites'),
        })
    }
}, {
    cardStyle: {
        backgroundColor: 'white'

    }
});

const AppSettingsTab = StackNavigator({
    AppSettings: {
        screen: AppSettings,
        path: '/',
        navigationOptions: ({navigation}) => ({
            title: I18n.t('settings')
        })
    },
    About: {
        screen: About,
        path: '/about',
        navigationOptions: ({navigation}) => ({
            title: I18n.t('about'),
            left: <TouchableOpacity activeOpacity={0.7}>
                <Label type={'body'} text={I18n.t('back')}/>
            </TouchableOpacity>
        })
    }
}, {
    headerMode: 'float',
    cardStyle:{
        backgroundColor:'white'
    }
});

const TabNav = TabNavigator({
  SearchTab: {
    screen: SearchTab,
    path: "search",
      navigationOptions: ({navigation}) => ({
          tabBarIcon: ({tintColor, focused}) => {
              return <Icon type='search' size={Settings.core.tabIconSize} active={focused}/>
          }
      })
  },
  FavoritesTab: {
      screen: FavoritesTab,
      path: "favorites",
      navigationOptions: ({navigation}) => ({
          tabBarIcon: ({tintColor, focused}) => {
              return <Icon type='favorite-on' size={Settings.core.tabIconSize} active={focused}/>
          }
      })
  },
  AppSettingsTab: {
    screen: AppSettingsTab,
    path: 'settings',
    navigationOptions: ({navigation}) => ({
      tabBarIcon: ({ tintColor, focused }) => (
        <Icon type='settings' size={Settings.core.tabIconSize} active={focused}/>
      ),
    })
  }
}, {
  initialRouteName: "SearchTab",
  cardStyle:{
      backgroundColor:'white'
  },
  lazy: true,
  tabBarOptions: {
    activeTintColor: Platform.OS === 'ios' ? 'red' : 'white',
    inactiveTintColor: Platform.OS === 'ios' ? 'black' : 'gray',
  },
});

const styles = StyleSheet.create({
  container: {
    marginTop: Platform.OS === 'ios' ? 20 : 0
  }
});

export default TabNav;

**Comment by spencercarli** [comment]
Unable to repo with this https://snack.expo.io/@spencercarli/[repro-attempt]-stacknavigator-inside-a-tabnavigator-causes-a-back-button-to-be-in-header

**Comment by kelset** [comment]
Pinging OP @benschel and @NunoCardoso since this issue is quite old and related to an old version of the lib (and Spencer's code can't reproduce). If no answer I'll close in 7 days.

**Comment by NunoCardoso** [comment]
Sorry for the late answer. 
I don't see it anymore, but i can't confirm why the issue went away, if it was package update, or anything else.

**Comment by NunoCardoso** [comment]
Sorry, it's still there :(

**Comment by kelset** [comment]
Can you please provide a repo that reproduces the issue, because @spencercarli couldn't reproduce it.

Once we have a repo / snack I'll reopen.

**Comment by NunoCardoso** [comment]
does a link like https://exp.host/@nunocardoso/{appname} is good enough?

I can share my AppNavigation.js file and a video of the iOS simulator.

**Comment by NunoCardoso** [comment]
Ok, I figured out what was happening. 

in My navReducer, I had:  
`const initialState = AppNavigator.router.getStateForAction(
     AppNavigator.router.getActionForPathAndParams('search')
);
`

Both my TabNavigator and StackNavigator had a 'search' entry, because I read somewhere that it helps. 

I renamed the 'search' on tab to 'searchtab', 'search' on stack to 'searchstack'. 
Added this code in the navReducer, based from what the OP had: 

```
const initialRoute = 'searchtab';

const initialState = AppNavigator.router.getStateForAction(
    NavigationActions.navigate({
        routeName: initialRoute
    })
);
```

And now it works. Not with 'searchstack', only with 'searchtab'. I will keep unique names.
